nmigen_boards/de4.py

```diff
@@ -0,0 +1,87 @@
+import os
+import subprocess
+
+from nmigen.build import *
+from nmigen.vendor.intel import *
+from .resources import *
+
+
+__all__ = ["DE4Platform"]
+
+#This is for the larger (500k LEs) chip. The DE4 also comes in a configuration with the EP4SGX230C2
+class DE4Platform(IntelPlatform):
+    device      = "EP4SGX530" # Stratix IV 530k LEs
+    package     = "KH40"     # FBGA-1517
+    speed       = "C2"
+    default_clk = "clk50"
+    default_rst = "rst"
+    resources   = [
+        Resource("rst", 0, PinsN("V34", dir="i"),
+                Attrs(io_standard="2.5-V")),
+
+        Resource("clk50", 0, Pins("AC35", dir="i"),
+                 Clock(50e6), Attrs(io_standard="2.5-V")),
+        Resource("clk50", 1, Pins("AV22", dir="i"),
+                 Clock(50e6), Attrs(io_standard="1.8-V")),
+        Resource("clk50", 2, Pins("AV19", dir="i"),
+                 Clock(50e6), Attrs(io_standard="1.8-V")),
+        Resource("clk50", 3, Pins("AC6", dir="i"),
+                 Clock(50e6), Attrs(io_standard="3.0-V PCI")),
+        Resource("clk50", 4, Pins("AB6", dir="i"),
+                 Clock(50e6), Attrs(io_standard="2.5-V")),
+        Resource("clk50", 5, Pins("A19", dir="i"),
+                 Clock(50e6), Attrs(io_standard="1.8-V")),
+        Resource("clk100", 0, Pins("A21", dir="i"),
+                 Clock(100e6), Attrs(io_standard="1.8-V")), #100MHz assumes SW7 is set to 00   
```

Comment:
Please make this configurable; see the Atlys board for an example of how to do so.